tests/end_to_end/tests.py

```diff
@@ -233,12 +239,11 @@ def test_run_configuration(self):
         signed_certs = self.step_acquire_certs(downloaded_csrs)
         self.apply_cert_config(signed_certs)
         self.step_cert_import()
-        self.step_cert_import()
         self.step_cert_register()
         self.step_cert_activate()
 
         # Wait for global configuration status updates
-        time.sleep(120)
+        waitfor(lambda: auth_cert_registration_global_configuration_update_received(self.config), 1, 300)
 
         self.step_subsystem_add_client()
         self.step_subsystem_register()
```

Comment:
E2E tests fail:
FAILED tests/end_to_end/tests.py::EndToEndTest::test_run_configuration - xrdsst.rest.rest.ApiException: (429)
Although when logging into the UI, it seems all the steps have been completed
When I comment out the last few lines from the tests:
        #self.step_autoconf()  # Idempotent

        #configured_servers_at_end = self.query_status()

        #assert_server_statuses_transitioned(unconfigured_servers_at_start, configured_servers_at_end)

then the tests pass.

Reply:
auto-configuration test still does not work:

    def test_status_uninitialized_server(self):
        with XRDSSTTest() as app:
            status_controller = StatusController()
            status_controller.app = app
            status_controller.load_config = (lambda: self.ss_config)
            status_controller._default()
    
            out, err = self.capsys.readouterr()
    
            # Global status
>           assert status_controller.app._last_rendered[0][1][0].count('FAIL') == 1
E           AssertionError: assert 0 == 1
E             +0
E             -1

tests/unit/test_status.py:105: AssertionError
--------------------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------------------

╒══════════╤════════════════╤═══════════╤════════╤════════╤═════════╤════════╤════════╤═════════╕
│ GLOBAL   │ SERVER         │ ROLES     │ INIT   │ TSAS   │ TOKEN   │ KEYS   │ CSRS   │ CERTS   │
╞══════════╪════════════════╪═══════════╪════════╪════════╪═════════╪════════╪════════╪═════════╡
│          │ longServerName │ NO ACCESS │        │        │         │        │        │         │
╘══════════╧════════════════╧═══════════╧════════╧════════╧═════════╧════════╧════════╧═════════╛


----------- coverage: platform linux, python 3.8.5-final-0 -----------
Name                              Stmts   Miss  Cover
-----------------------------------------------------
xrdsst/controllers/auto.py           58      3    95%
xrdsst/controllers/base.py          299     13    96%
xrdsst/controllers/cert.py          190      8    96%
xrdsst/controllers/client.py         91      5    95%
xrdsst/controllers/init.py           62      0   100%
xrdsst/controllers/service.py       102      5    95%
xrdsst/controllers/status.py         53      3    94%
xrdsst/controllers/timestamp.py     101      7    93%
xrdsst/controllers/token.py         163     10    94%
-----------------------------------------------------
TOTAL                              1119     54    95%
Coverage HTML written to dir coverage-report

======================================================================================================= short test summary info ========================================================================================================
FAILED tests/integration/test_xrdsst_auto.py::TestXrdsstAuto::test_run_auto_configuration - AssertionError: assert 2 == 1
FAILED tests/unit/test_status.py::TestStatus::test_status_uninitialized_server - AssertionError: assert 0 == 1